src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -68,48 +69,101 @@ public IList<ISourceReference> GetSources()
                                      .ToList();
         }
 
-        public Assembly Load(IAssemblyLoadContext loadContext)
+        public EmitResult EmitResourceAssembly(
+            AssemblyName assemblyName,
+            IEnumerable<ResourceDescriptor> resourceDescriptors,
+            AfterCompileContext afterCompileContext)
         {
-            using (var pdbStream = new MemoryStream())
-            using (var assemblyStream = new MemoryStream())
-            {
-                IList<ResourceDescription> resources = CompilationContext.Resources;
+            var resources = resourceDescriptors
+                .Select(res => new ResourceDescription(res.Name, res.StreamFactory, isPublic: true));
 
-                Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
+            var mainCompilation = afterCompileContext.Compilation;
+            afterCompileContext.Compilation = CSharpCompilation.Create(
```

Comment:
Why is this code still using the AfterCompileContext even though, we're not calling the ICompileModule anymore